core/src/main/java/eu/tsystems/mms/tic/testframework/utils/PdfUtils.java

```diff
@@ -77,25 +76,231 @@ public static String getStringFromPdf(String pdffilelocation) {
      * @return PDF content as String
      */
     public static String getStringFromPdf(InputStream stream) {
-        PDDocument pdDoc = null;
-        PDFTextStripper stripper = null;
-        String parsedtext = null;
+        PDDocument pdDoc = getPdDocument(stream);
+        PDFTextStripper stripper;
+        String parsedText;
         try {
-            pdDoc = PDDocument.load(stream);
             stripper = new PDFTextStripper();
-            parsedtext = stripper.getText(pdDoc);
-            return parsedtext;
+            parsedText = stripper.getText(pdDoc);
+            return parsedText;
         } catch (IOException e) {
-            LOG.error("Error reading pdf file", e);
-            return "";
+            throw new RuntimeException("Error reading pdf file", e);
         } finally {
-            try {
-                if (pdDoc != null) {
-                    pdDoc.close();
-                }
-            } catch (IOException e) {
-                LOG.debug("error closing pdf stream", e);
+            PdfUtils.closeDocument(pdDoc);
+        }
+    }
+
+    /**
+     * Get content of one specific page of pdf as string.
+     *
+     * @param pdfFileLocation Absolute path of file
+     * @param pageNumber The number of the page
+     * @return PDF page content as String
+     */
+    public static String getStringFromPdf(String pdfFileLocation, int pageNumber) {
+        File pdfFile = new File(pdfFileLocation);
+        FileInputStream stream = PdfUtils.getFileInputStream(pdfFile);
+        return getStringFromPdf(stream, pageNumber);
+    }
+
+    /**
+     * Get content of one specific page of pdf as string.
+     *
+     * @param stream File stream
+     * @param pageNumber The number of the page
+     * @return PDF page content as String
+     */
+    public static String getStringFromPdf(InputStream stream, int pageNumber) {
+        PDDocument pdDoc = getPdDocument(stream);
+        PdfUtils.checkPageNumber(pdDoc, pageNumber);
+
+        PDFTextStripper stripper = new PDFTextStripper();
+        stripper.setStartPage(pageNumber);
+        stripper.setEndPage(pageNumber);
+        try {
+            return stripper.getText(pdDoc);
+        } catch (IOException e) {
+            throw new RuntimeException("Error reading pdf file", e);
+        } finally {
+            PdfUtils.closeDocument(pdDoc);
+        }
+    }
+
+    /**
+     * Create an image for each page of pdf.
+     *
+     * @param pdfFileLocation Absolute path of file
+     * @param dpi The resolution of the rendered image
+     * @return A list of rendered image files
+     */
+    public static List<File> pdfToImage(String pdfFileLocation, int dpi) {
```

Comment:
Please rename the method to `getImageFromPdf` to be consistent with `getStringFromPdf`